applications/asset_tracker_v2/tests/location_module/src/location_module_test.c

```diff
@@ -0,0 +1,554 @@
+	bool ret = LOCATION_MODULE_EVT_HANDLER((struct app_event_header *)app_module_event);
+
+	app_event_manager_free(app_module_event);
+
+	TEST_ASSERT_EQUAL(0, ret);
+}
+
+static void setup_location_module_in_running_state(void)
+{
+	setup_location_module_in_init_state();
+
+	/* Send MODEM_EVT_INITIALIZED. */
+	__wrap_location_init_ExpectAndReturn(&location_event_handler, 0);
+
+	__wrap_app_event_manager_alloc_ExpectAnyArgsAndReturn(&modem_module_event_memory);
+	__wrap_app_event_manager_free_ExpectAnyArgs();
+	struct modem_module_event *modem_module_event = new_modem_module_event();
+
+	modem_module_event->type = MODEM_EVT_INITIALIZED;
+
+	bool ret = LOCATION_MODULE_EVT_HANDLER((struct app_event_header *)modem_module_event);
+
+	app_event_manager_free(modem_module_event);
+
+	/* Send DATA_EVT_CONFIG_INIT. */
+	__wrap_app_event_manager_alloc_ExpectAnyArgsAndReturn(&data_module_event_memory);
+	__wrap_app_event_manager_free_ExpectAnyArgs();
+	struct data_module_event *data_module_event = new_data_module_event();
+
+	data_module_event->type = DATA_EVT_CONFIG_INIT;
+	data_module_event->data.cfg.location_timeout = 30;
+	data_module_event->data.cfg.no_data.gnss = false;
+	data_module_event->data.cfg.no_data.neighbor_cell = false;
+
+	ret = LOCATION_MODULE_EVT_HANDLER((struct app_event_header *)data_module_event);
+
+	app_event_manager_free(data_module_event);
+
+	TEST_ASSERT_EQUAL(0, ret);
+}
+
+/* Set location module into state where location_request() has been called.
+ */
+static void setup_location_module_in_active_state(void)
+{
+	/* Pre-condition. */
+	setup_location_module_in_running_state();
+
+	/* Set up expectations.  */
+	/* Set callback to validate location module events. */
+	__wrap__event_submit_Stub(&validate_location_module_evt);
+
+	/* Set default location configuration. */
+	enum location_method methods[] = { LOCATION_METHOD_GNSS, LOCATION_METHOD_CELLULAR };
+
+	struct location_config config_defaults = {
+		.methods_count = 2,
+		.interval = 0,
+		.timeout = 300000,
+		.mode = LOCATION_REQ_MODE_FALLBACK,
+		.methods = {
+			{.method = LOCATION_METHOD_GNSS, .gnss.timeout = 120000},
+			{.method = LOCATION_METHOD_CELLULAR, .cellular.timeout = 30000}
+		}
+	};
+	__wrap_location_config_defaults_set_Expect(NULL, 2, methods);
+	__wrap_location_config_defaults_set_IgnoreArg_config();
+	__wrap_location_config_defaults_set_ReturnThruPtr_config(&config_defaults);
+
+	/* Location configuration modified by location module for location request. */
+	struct location_config config_location_request = {
+		.methods_count = 2,
+		.interval = 0,
+		.timeout = 300000,
+		.mode = LOCATION_REQ_MODE_FALLBACK,
+		.methods = {
+			{.method = LOCATION_METHOD_GNSS, .gnss.timeout = 30000},
+			{.method = LOCATION_METHOD_CELLULAR, .cellular.timeout = 11000}
+		}
+	};
+	__wrap_location_request_ExpectAndReturn(&config_location_request, 0);
+	__wrap_location_request_IgnoreArg_config(); /* TODO: REMOVE */
```

Comment:
Eventually no. I have similar things in Location library tests as the initial tests didn't verify everything to get something tested "quickly". I'm still trying to see if I can figure out why this IgnoreArg is needed but if I get necessary approvals and the moon is in the right position to get CI through, I wouldn't like to hold the merging because of this.